homeassistant/components/backup/manager.py

```diff
@@ -244,12 +244,21 @@ async def _async_upload_backup(
     ) -> None:
         """Upload a backup to selected agents."""
         LOGGER.warning("Uploading backup %s to agents %s", backup.backup_id, agent_ids)
+
+        async def send_backup() -> AsyncGenerator[bytes]:
+            with await self.hass.async_add_executor_job(path.open, "rb") as f:
```

Comment:
`with await` seems weird. Normally it's `async with` in async context.

Reply:
`async with` means enter an async context manager, that's not what's happening here.

Maybe it would be clearer to instead do:
```py
f = await self.hass.async_add_executor_job(path.open, "rb")
with f:
    ...
```